walk-in-interview/src/main/java/com/google/job/servlets/JobsListingsServlet.java

```diff
@@ -0,0 +1,229 @@
+        String maxLimitStr = ServletUtils.getStringParameter(request, MAX_LIMIT_PARAM, /* defaultValue= */ "");
+
+        if (maxLimitStr.isEmpty()) {
+            throw new IllegalArgumentException("max limit param should not be empty");
+        }
+
+        try {
+            return Integer.parseInt(maxLimitStr);
+        } catch(NumberFormatException e) {
+            throw new IllegalArgumentException("max limit param should be an int");
+        }
+    }
+
+    /**
+     * Returns the region as a SingaporeRegion enum.
+     *
+     * @param request From the GET request.
+     * @return SingaporeRegion enum given the id in the request params.
+     * @throws IllegalArgumentException if the id is invalid.
+     */
+    private static SingaporeRegion parseRegion(HttpServletRequest request) throws IllegalArgumentException {
+        String region = ServletUtils.getStringParameter(request, REGION_PARAM, /* defaultValue= */ "");
+
+        if (region.isEmpty()) {
+            throw new IllegalArgumentException("region param should not be empty");
+        }
+
+        return SingaporeRegion.getFromId(region); // Illegal Argument Exception may be thrown
+    }
+
+    /**
+     * Returns the sorting as a Filter enum.
+     *
+     * @param request From the GET request.
+     * @return Filter enum given the id in the request params.
+     * @throws IllegalArgumentException if the id is invalid.
+     */
+    public static Filter parseSortBy(HttpServletRequest request) throws IllegalArgumentException {
+        String sortById = ServletUtils.getStringParameter(request, SORT_BY_PARAM, /* defaultValue= */ "");
+
+        if (sortById.isEmpty()) {
+            throw new IllegalArgumentException("sort by param should not be empty");
+        }
+
+        return Filter.getFromId(sortById); // Illegal Argument Exception may be thrown
+    }
+
+    /**
+     * Returns the ordering as an Order enum.
+     *
+     * @param request From the GET request.
+     * @return Order enum given the id in the request params.
+     * @throws IllegalArgumentException if the id is invalid.
+     */
+    public static Order parseOrder(HttpServletRequest request) throws IllegalArgumentException {
+        String orderId = ServletUtils.getStringParameter(request, ORDER_PARAM, /* defaultValue= */ "");
+
+        if (orderId.isEmpty()) {
+            throw new IllegalArgumentException("order param should not be empty");
+        }
+
+        return Order.getFromId(orderId); // Illegal Argument Exception may be thrown
+    }
+
+    /**
+     * Returns the page size as an int.
+     *
+     * @param request From the GET request.
+     * @return the page size.
+     * @throws IllegalArgumentException if the page size is invalid.
+     */
+    public static int parsePageSize(HttpServletRequest request) throws IllegalArgumentException {
+        String pageSizeStr = ServletUtils.getStringParameter(request, PAGE_SIZE_PARAM, /* defaultValue= */ "");
```

Comment:
Would it be better to default to some value instead of throw? Or are we currently trying to be explicit in all parameters?

Reply:
I think I was trying to be explicit with all the parameters. I will set the region to ENTIRE is there is no region though.

Reply:
Currently pageSize and pageIndex are the default values though. Since pagination hasn't been implemented yet